dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/document/visibility/DocumentVisibility.java

```diff
@@ -0,0 +1,584 @@
+      final DocumentVisibilityParser p = new DocumentVisibilityParser();
+      node = p.parse(expression);
+    } else {
+      node = EMPTY_NODE;
+    }
+    this.expression = expression;
+  }
+
+  /**
+   * Creates an empty visibility. Normally, elements with empty visibility can be seen by everyone. Though, one could change this behavior with filters.
+   *
+   * @see #DocumentVisibility(String)
+   */
+  public DocumentVisibility() {
+    this(new byte[] {});
+  }
+
+  /**
+   * Creates a document visibility for a Mutation.
+   *
+   * @param expression
+   *          An expression of the rights needed to see this mutation. The expression syntax is defined at the class-level documentation
+   */
+  public DocumentVisibility(final String expression) {
+    this(expression.getBytes(UTF_8));
+  }
+
+  /**
+   * Creates a document visibility for a Mutation.
+   *
+   * @param expression
+   *          visibility expression
+   * @see #DocumentVisibility(String)
+   */
+  public DocumentVisibility(final Text expression) {
+    this(TextUtil.getBytes(expression));
+  }
+
+  /**
+   * Creates a document visibility for a Mutation from a string already encoded in UTF-8 bytes.
+   *
+   * @param expression
+   *          visibility expression, encoded as UTF-8 bytes
+   * @see #DocumentVisibility(String)
+   */
+  public DocumentVisibility(final byte[] expression) {
+    validate(expression);
+  }
+
+  @Override
+  public String toString() {
+    return "[" + new String(expression, UTF_8) + "]";
+  }
+
+  /**
+   * See {@link #equals(DocumentVisibility)}
+   */
+  @Override
+  public boolean equals(final Object obj) {
+    if (obj instanceof DocumentVisibility) {
+		return equals(obj);
+	}
+    return false;
+  }
+
+  /**
+   * Compares two DocumentVisibilities for string equivalence, not as a meaningful comparison of terms and conditions.
+   *
+   * @param otherLe
+   *          other document visibility
+   * @return true if this visibility equals the other via string comparison
+   */
+  public boolean equals(final DocumentVisibility otherLe) {
+    return Arrays.equals(expression, otherLe.expression);
+  }
+
+  @Override
+  public int hashCode() {
+    return Arrays.hashCode(expression);
+  }
+
+  /**
+   * Gets the parse tree for this document visibility.
+   *
+   * @return parse tree node
+   */
+  public Node getParseTree() {
+    return node;
+  }
+
+  /**
+   * Properly quotes terms in a document visibility expression. If no quoting is needed, then nothing is done.
+   *
+   * <p>
+   * Examples of using quote :
+   *
+   * <pre>
+   * import static org.apache.rya.mongodb.document.visibility.DocumentVisibility.quote;
+   *   .
+   *   .
+   *   .
+   * DocumentVisibility dv = new DocumentVisibility(quote(&quot;A#C&quot;) + &quot;&amp;&quot; + quote(&quot;FOO&quot;));
+   * </pre>
+   *
+   * @param term
+   *          term to quote
+   * @return quoted term (unquoted if unnecessary)
+   */
+  public static String quote(final String term) {
```

Comment:
why not create a new class level String of a UTF_8 quote?  UTF_8_QUOTE + "some auth" + UTF_8_QUOTE